src/Common/src/Interop/Unix/System.Native/Interop.Dup.cs

```diff
@@ -1,13 +1,15 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
 using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
 
 internal static partial class Interop
 {
     internal static partial class Sys
     {
         [DllImport(Libraries.SystemNative, SetLastError = true)]
-        internal static extern int Dup(int oldfd);
+        internal static extern IntPtr Dup(SafeFileHandle oldfd);
```

Comment:
Should this return a SafeFileHandle instead of an IntPtr?  The SafeFileHandle.Open would then change to use the CheckIo function I suggested that takes a SafeHandle.